run_so2sat_experiments.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Runs the train script with a grid of hyperparameters."""
+import itertools
+import os
+import subprocess
+from multiprocessing import Process, Queue
+
+# list of GPU IDs that we want to use, one job will be started for every ID in the list
+GPUS = [0, 1, 2, 3, 4, 6]
+TEST_MODE = False  # if False then print out the commands to be run, if True then run
+
+# Hyperparameter options
+model_options = ["resnet18", "resnet152"]
+# lr_options = [1e-2, 1e-3, 1e-4]
+loss_options = ["ce"]
+weight_options = ["imagenet_only", "random", "imagenet_and_random", "random_rgb"]
+
+
+def do_work(work: "Queue[str]", gpu_idx: int) -> bool:
+    """Process for each ID in GPUS."""
+    while not work.empty():
+        experiment = work.get()
+        experiment = experiment.replace("GPU", str(gpu_idx))
+        print(experiment)
+        if not TEST_MODE:
+            subprocess.call(experiment.split(" "))
+    return True
+
+
+def main() -> None:
+    """Main."""
+    work: Queue[str] = Queue()
+
+    for (model, loss, weights) in itertools.product(
+        model_options,
+        # lr_options,
+        loss_options,
+        weight_options,
+    ):
+
+        # experiment_name = f"{model}_{lr}_{loss}_{weights.replace('_','-')}_randomerasing"
+        experiment_name = f"{model}_autolr_{loss}_{weights.replace('_','-')}"
+
+        output_dir = "output/so2sat_experiments/"
+
+        if not os.path.exists(os.path.join(output_dir, experiment_name)):
+
+            command = (
+                "python train.py"
+                + " config_file=conf/so2sat.yaml"
+                + f" experiment.name={experiment_name}"
+                + f" experiment.module.classification_model={model}"
+                # + f" experiment.module.learning_rate={lr}"
+                + " trainer.auto_lr_find=True"
+                + f" experiment.module.loss={loss}"
+                + f" experiment.module.weights={weights}"
+                + f" experiment.datamodule.weights={weights}"
+                + f" program.output_dir={output_dir}"
+                + f" program.log_dir={output_dir}/logs"
```

Comment:
Are only strings accepted for this? This seems vulnerable to injection attacks.

Reply:
I don't think this is relevant, this is a script for running experiments